tf/terraform.go

```diff
@@ -86,21 +82,27 @@ func ParseModules(path string) ([]Module, error) {
 		if len(block.Labels) == 1 {
 			moduleName = block.Labels[0]
 		} else {
-			errs.Append(errors.E(ErrTerraformSchema, block.OpenBraceRange,
-				"\"module\" block must have 1 label"))
+			logger.Debug().Msgf("ignoring module block with %d labels", len(block.Labels))
+
+			continue
 		}
+		logger = logger.With().
+			Str("module", moduleName).
+			Logger()
 
 		logger.Trace().Msg("Get source attribute.")
 		source, ok, err := findStringAttr(block, "source")
 		if err != nil {
-			errs.Append(errors.E(ErrTerraformSchema, err,
-				"looking for module.%q.source attribute", moduleName))
+			logger.Debug().
+				Err(err).
+				Msg("ignoring module block without source")
+
+			continue
 		}
 		if !ok {
-			errs.Append(errors.E(ErrTerraformSchema,
-				hcl.RangeBetween(block.OpenBraceRange, block.CloseBraceRange),
-				"module must have a \"source\" attribute",
-			))
+			logger.Debug().Msg("ignoring module block without source")
+
+			continue
```

Comment:
I think we had a bug here, it was appending the error on the err list but also adding the module with an empty source on the result.

Reply:
yeah, luckily this didn't lead to a real bug because all the code using the module list is guarded by checking the source:
https://github.com/mineiros-io/terramate/blob/main/manager.go#L463
https://github.com/mineiros-io/terramate/blob/main/modvendor/download/download.go#L273

thanks for fixing this!